src/views/PopularRepositories.vue

```diff
@@ -1,20 +1,64 @@
 <template>
   <div>
     <h3 class="center-align">Popular Repositories</h3>
-    <slot />
-    <Repositories />
+    <p class="center">
+      Showing <strong>{{ repositories.length.toLocaleString() }}</strong> repositories
+      <span>sorted by stars.</span>
+    </p>
+    <InputSearch label="Filter repository by name or description..." v-model="searchTerm" />
+    <Repositories v-for="repository in filteredRepositories" :repository="repository" :key="repository.id" />
   </div>
 </template>
 
 <script>
+import { mapGetters } from "vuex";
+import * as JsSearch from "js-search";
+import sortBy from "lodash/sortBy";
 import Repositories from "@/components/RepositoryCard";
+import InputSearch from "@/components/InputSearch";
+const searcher = new JsSearch.Search("position");
+searcher.indexStrategy = new JsSearch.AllSubstringsIndexStrategy();
+searcher.searchIndex = new JsSearch.UnorderedSearchIndex();
+searcher.addIndex("name");
+searcher.addIndex("description");
 
 export default {
   name: "PopularRepositories",
   components: {
-    Repositories
+    Repositories,
+    InputSearch
+  },
+  data() {
+    return {
+      searchTerm: ""
+    };
+  },
+  computed: {
+    ...mapGetters({
+      repositories: "Repositories/repositories"
+    }),
+    filteredRepositories() {
+      if (!this.searchTerm) {
+        return this.repositories.slice(0, 10);
```

Comment:
Why aren't we sorting by stargazers here already? I'm unaware of the intent here.

Reply:
@vyzaldysanchez Because now the data from the API is already sorted by stargazers.

Reply:
BTW this `slice` should be removed once you implement the virtual scroll

Reply:
> @vyzaldysanchez Because now the data from the API is already sorted by stargazers.

You have removed it in e5656f755f1c18732463838c22977472bd5ab607